plugins/woocommerce/src/Internal/Admin/Settings/PaymentProviders/WooPayments/WooPaymentsRestController.php

```diff
@@ -0,0 +1,920 @@
+	 * @return WP_Error|WP_REST_Response
+	 */
+	protected function get_onboarding_details( WP_REST_Request $request ) {
+		$location = $request->get_param( 'location' );
+		if ( empty( $location ) ) {
+			// Fall back to the providers country if no location is provided.
+			$location = $this->payments->get_country();
+		}
+
+		try {
+			$onboarding_details = $this->woopayments->get_onboarding_details( $location, $this->get_rest_url_path( 'onboarding' ) );
+		} catch ( Exception $e ) {
+			return new WP_Error( 'woocommerce_rest_woopayments_onboarding_error', $e->getMessage(), array( 'status' => 500 ) );
+		}
+
+		return rest_ensure_response( $this->prepare_onboarding_details_response( $onboarding_details ) );
+	}
+
+	/**
+	 * Handle the onboarding step start action.
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 *
+	 * @return WP_Error|WP_REST_Response The response.
+	 */
+	protected function handle_onboarding_step_start( WP_REST_Request $request ) {
+		$step_id = $request->get_param( 'step' );
+		if ( empty( $step_id ) || ! $this->woopayments->is_valid_onboarding_step_id( $step_id ) ) {
+			return new WP_Error( 'woocommerce_rest_woopayments_onboarding_invalid_step', __( 'Invalid onboarding step ID.', 'woocommerce' ), array( 'status' => 400 ) );
+		}
+
+		$location = $request->get_param( 'location' );
+		if ( empty( $location ) ) {
+			// Fall back to the providers country if no location is provided.
+			$location = $this->payments->get_country();
+		}
+
+		$previous_status = $this->woopayments->get_onboarding_step_status( $step_id, $location );
+
+		try {
+			$this->woopayments->set_onboarding_step_started( $step_id, $location );
+		} catch ( Exception $e ) {
+			return new WP_Error( 'woocommerce_rest_woopayments_onboarding_error', $e->getMessage(), array( 'status' => 500 ) );
+		}
+
+		$response = array(
+			'success'         => true,
+			'previous_status' => $previous_status,
+			'current_status'  => $this->woopayments->get_onboarding_step_status( $step_id, $location ),
+		);
+
+		return rest_ensure_response( $response );
+	}
+
+	/**
+	 * Handle the onboarding step save action.
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 *
+	 * @return WP_Error|WP_REST_Response The response.
+	 */
+	protected function handle_onboarding_step_save( WP_REST_Request $request ) {
+		$step_id = $request->get_param( 'step' );
+		if ( empty( $step_id ) || ! $this->woopayments->is_valid_onboarding_step_id( $step_id ) ) {
+			return new WP_Error( 'woocommerce_rest_woopayments_onboarding_invalid_step', __( 'Invalid onboarding step ID.', 'woocommerce' ), array( 'status' => 400 ) );
+		}
+
+		$location = $request->get_param( 'location' );
+		if ( empty( $location ) ) {
+			// Fall back to the providers country if no location is provided.
+			$location = $this->payments->get_country();
+		}
+
+		try {
+			$this->woopayments->onboarding_step_save( $step_id, $location, $request->get_params() );
```

Comment:
I've seen we are trying to access request data, so we should pass the body instead
```suggestion
			$this->woopayments->onboarding_step_save( $step_id, $location, json_decode( $request->get_body(), true ) );
```

Reply:
I used the very broad `$request->get_params()` since it will gather all the data, regardless of the transport (query params, body). Do you see something off with this approach?

The service save will pick only the data it can handle, per-step.